action-processor/assign-role-action.js

```diff
@@ -29,16 +29,22 @@ async function createRole (namespace, action) {
     let roleData = action.data
     let users = roleData.users
     let orgIdLabel = roleData.orgId
-    let savedOrgAction = localStorage.getItem(namespace, orgIdLabel)
-    let response = savedOrgAction.response
-    let orgId = response.Id
+
+    let savedOrgAction = await localStorage.getItem(namespace, orgIdLabel)
+
+    if(!(savedOrgAction && savedOrgAction.response)){
+        savedOrgAction = await localStorage.getItem('global', orgIdLabel)
+    }
+    let orgId = savedOrgAction.response.Id
 
     var userList = []
-    let userNameSpace = action.global == 'true' ? 'global' : namespace
     for (const userLabel of users) {
-        let savedAction = await localStorage.getItem(userNameSpace, userLabel)
+        let savedAction = await localStorage.getItem(namespace, userLabel)
+        if(!savedAction)
+            savedAction = await localStorage.getItem('global', userLabel)
```

Comment:
Could you put the code which fails back to using global into a single function ? As its done in 2 places

JS pseudo code...

function getItemWithGlobalFallBack(etc.....)
{
  let savedAction = await localStorage.getItem(namespace, userLabel)
         if(!savedAction)
             savedAction = await localStorage.getItem('global', userLabel)
return ....
}

Reply:
done